Makefile

```diff
@@ -83,15 +78,17 @@ else
 	mv ./lib/pyinstall/lib/$(PYTHON_LIB) $@
 	mv ./lib/pyinstall/lib/python$(PYTHON_VER_MAJOR) ./lib/.
 	rm -rf ./lib/pyinstall
-endif
 
 ./obj/%.o: ./src/%.c
 	mkdir -p $(dir $@)
 	$(CC) -MT $@ -MMD -MP -MF ./obj/$*.d $(CFLAGS) $(DEFS) -c $< -o $@
 
 pf: $(PF_OBJS)
 	mkdir -p ./bin
-	$(CC) $? -o $(BIN) $(LDFLAGS)
+	$(CC) $? -o $(BIN) $(LDFLAGS) $(PLATFORM_LDFLAGS)
+ifeq ($(UNAME),Darwin)
+	install_name_tool -change /usr/local/lib/libSDL2-2.0.0.dylib $(CURDIR)/lib/libSDL2.dylib -change /usr/local/lib/libGLEW.2.1.0.dylib $(CURDIR)/lib/libGLEW.dylib bin/pf
```

Comment:
Ok, here is the most important comment: 

Here I believe that the absolute path of the built shared library is being referenced. This is undesirable - the shared library should still be correctly located even if the entire engine folder is moved somewhere else. This is the purpose of the `rpath` option to the linker. The point is that the directory `../lib/` relative to the binary being run will be searched for shared libraries which can satisfy the binary's dependencies. 

I think that what is happening here is that when the linker is passed the options, `-lSDL2` and `-lGLEW`, it links against shared libraries installed on the system. Due to this, there is a need to use `install_name_tool` to patch the dependency paths. 

Can you try the following:
Forcefully link to the locally-compiled libraries using `-l:$(SDL2_LIB) -l:$(GLEW_LIB) -l:$(PYTHON_LIB)` as on Linux and see if it's possible to avoid the `install_name_tool` step?

As well, that would remove the need for having different `SDL2_LIB` and `SDL2_LIB_ORIG` variables. To explain, the reason that the shared library names in the `../lib` folder are different from the original names that are output by the build process is that I tried to make them match the names expected by the linker. The idea was that when the linker was passed generic linker flags like `-lSDL2`, it would first try to find the shared library dependencies in `../lib` and, if they were not there, it would fall back to installed system libraries. In practice, this doesn't work with my current Makefile and I've already resorted to forcefully linking the locally built shared libraries on Linux. The different names is kind of just useless legacy baggage and can be removed as well.